src/tracks/gosling-track/gosling-track.ts

```diff
@@ -801,12 +802,24 @@ const factory: PluginTrackFactory<Tile, GoslingTrackOptions> = (HGC, context, op
         }
 
         /**
-         * Creates an array of SingleTracks if there are overlaid tracks
+         * Creates an array of SingleTracks if there are overlaid tracks.
+         * This method cannot be private because it is called by functions which are called by super.draw();
          */
-        #getResolvedTracks() {
-            const copy = structuredClone(this.options.spec);
+        getResolvedTracks() {
+            if (!this.resolvedTracks) {
+                const copy = structuredClone(this.options.spec);
+                const tracks = resolveSuperposedTracks(copy).filter(t => t.mark !== 'brush');
+                // We will never need to access the values field in the data spec. It can be quite large which can degrade performance so we remove it.
+                tracks.forEach(track => {
+                    if ('values' in track.data) {
+                        track.data.values = [];
+                    }
+                });
+                this.resolvedTracks = tracks;
+            }
```

Comment:
Rather than creating a resolved spec every time, we make it once and reuse it.

Reply:
I tried to come up with ways in creating a new resolved spec at every render is needed but couldn't think of one. Sehi do you know if there are any?

Reply:
It is possible that the track spec can be edited by users even after the track has been rendered, so we would need to force updating `resolvedTracks` inside the `override rerender(newOptions: GoslingTrackOptions)` function. This is the function that is called when a user edits a track spec. I don't think this will decrease the performance though since this is called only when user edits the spec.

Reply:
That's exactly what I was looking for, thank you! Just changed.